index.bs

```diff
@@ -4289,106 +4333,122 @@ partial interface MLGraphBuilder {
-
-    // input gate (i)
-    let i = builder.sigmoid(builder.add(
-      builder.mul(
-        cellState,
-        (options.peepholeWeight ?
-           builder.slice(options.peepholeWeight, [0], [hiddenSize]) :
-           zero)),
-      builder.add(
+    function lstmCell(
+      builder,
+      input,
+      weight,
+      recurrentWeight,
+      hiddenState,
+      cellState,
+      hiddenSize,
+      options) {
+      const zero = builder.constant(input.dataType, 0);
+
+      const inputSize = input.shape()[1];
+
+      // input gate (i)
+      let i = builder.sigmoid(builder.add(
+        builder.mul(
+          cellState,
+          (options.peepholeWeight ?
+             builder.slice(options.peepholeWeight, [0], [hiddenSize]) :
+             zero)),
         builder.add(
-          (options.bias ? builder.slice(options.bias, [0], [hiddenSize]) : zero),
-          (options.recurrentBias ?
-             builder.slice(options.recurrentBias, [0], [hiddenSize]) :
+          builder.add(
+            (options.bias ? builder.slice(options.bias, [0], [hiddenSize]) : zero),
+            (options.recurrentBias ?
+               builder.slice(options.recurrentBias, [0], [hiddenSize]) :
+               zero)),
+          builder.add(
+            builder.matmul(
+              input,
+              builder.transpose(
+                builder.slice(weight, [0, 0], [hiddenSize, inputSize]))),
+            builder.matmul(
+              hiddenState,
+              builder.transpose(builder.slice(
+                recurrentWeight, [0, 0], [hiddenSize, hiddenSize])))))));
+
+      // forget gate (f)
+      let f = builder.sigmoid(builder.add(
+        builder.mul(
+          cellState,
+          (options.peepholeWeight ?
+             builder.slice(options.peepholeWeight, [2 * hiddenSize], [hiddenSize]) :
              zero)),
         builder.add(
-          builder.matmul(
-            input,
-            builder.transpose(
-              builder.slice(weight, [0, 0], [hiddenSize, inputSize]))),
-          builder.matmul(
-            hiddenState,
-            builder.transpose(builder.slice(
-              recurrentWeight, [0, 0], [hiddenSize, hiddenSize])))))));
-
-    // forget gate (f)
-    let f = builder.sigmoid(builder.add(
-      builder.mul(
-        cellState,
-        (options.peepholeWeight ?
-           builder.slice(options.peepholeWeight, [2 * hiddenSize], [hiddenSize]) :
-           zero)),
-      builder.add(
+          builder.add(
+            (options.bias ?
+               builder.slice(options.bias, [2 * hiddenSize], [hiddenSize]) :
+               zero),
+            (options.recurrentBias ?
+               builder.slice(
+                 options.recurrentBias, [2 * hiddenSize], [hiddenSize]) :
+               zero)),
+          builder.add(
+            builder.matmul(
+              input,
+              builder.transpose(builder.slice(
+                weight, [2 * hiddenSize, 0], [hiddenSize, inputSize]))),
```

Comment:
This first-parameter-hugging makes it a little murkier the indentation levels when I'm visually trying to correspond the beginning and ending of things (yes, it saves some digital ink, but code isn't really printed on trees anymore). I'll look for some options to configure that... (but resolve me)